### PR TITLE
[WIP]docs: add documentation for ngNonBindable

### LINKED ISSUE #19497: ngNonBindable and ngNoForm are not in the docs
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

Searching docs for `ngNonBindable` or `ngNoForm` shows no results.

## Expected behavior

It should work.

## What is the motivation / use case for changing the behavior?

Finding it :smile:

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: 
Closes #19497

## What is the new behavior?

A new short section in Attribute Binding explaining `ngNonBindable` and an addition to the Attribute Binding example to demo it.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/attribute-directives.md [hunk 1]
```diff
@@ -299,14 +299,29 @@ and fall back to "violet" as the default color.
...
 </figure>
 
+<hr />
+
+{@a ngNonBindable}
+
+## `ngNonBindable`
+
+The built-in attribute directive `ngNonBindable` allows you to render code in
+your template. With `ngNonBindable`, Angular won't evaluation expressions in
```
Comment (JoostK) [inline aio/content/guide/attribute-directives.md]: evaluation -> evaluate

Comment (JoostK) [reply aio/content/guide/attribute-directives.md]: Also, could we clarify if this only affects children, or also attribute bindings and directive matching? I believe those are unactivated as well.

File: aio/content/guide/attribute-directives.md [hunk 2]
```diff
@@ -299,14 +299,29 @@ and fall back to "violet" as the default color.
...
   <img src="generated/images/guide/attribute-directives/highlight-directive-final-anim.gif" alt="Final Highlight">
 </figure>
 
+<hr />
+
+{@a ngNonBindable}
+
+## `ngNonBindable`
+
+The built-in attribute directive `ngNonBindable` allows you to render code in
```
Comment (trotyl) [inline aio/content/guide/attribute-directives.md]: `ngNonBindable` is not a directive, but a template primitive like `ng-template` and `ng-content`.

File: aio/content/guide/attribute-directives.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/attribute-directives.md]: Oh, thanks @JoostK for your review and for catching that typo. As for clarification, definitely! I'm going to have to research/experiment, but those are great points. I'll update as soon as I've figured out the details.

File: aio/content/guide/attribute-directives.md (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/guide/attribute-directives.md]: Thank you @trotyl. Would this be more accurate?

> The built-in template primitive...

And do you think it works here in the attribute directives doc? EDIT: as in maybe it should go elsewhere?

Comment (kapunahelewong) [reply aio/content/guide/attribute-directives.md]: And maybe I should mention it looks like an attribute directive but isn't?

Comment (trotyl) [reply aio/content/guide/attribute-directives.md]: Technically I think it belongs to `template syntax`, but I also prefer to having their own API entries for all keyword-based primitives (`ng-template`, `ng-container`, `ng-content`, `[ngNonBindable]`).

Comment (kapunahelewong) [reply aio/content/guide/attribute-directives.md]: Thanks @trotyl 

Hey @brandonroberts and @jenniferfell what are your thoughts on where this ought to go?

File: aio/content/guide/attribute-directives.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/attribute-directives.md]: Hey @JoostK, can you explain what directive matching is, or point me in the right direction? I searched but didn't find anything. I've been experimenting though: [https://stackblitz.com/edit/ngnonbindable](https://stackblitz.com/edit/ngnonbindable)

Comment (JoostK) [reply aio/content/guide/attribute-directives.md]: Hey @kapunahelewong that is great experiment :) It shows almost everything there is to know about `ngNonBindable`.

What I mean by directive matching is the process of figuring out which directives to apply to an element based on its tag/attributes. The main question is whether or not having `ngNonBindable` on an element would exclude that element from directive matching.

Your examples show that the full descendants tree is excluded from directive matching: no directives are instantiated on the elements even though an element would be applicable to the `appHighlight`, seen by the fact that the `appHighlight` elements within `ngNonBindable` don't work. Interestingly, if you were to move `ngNonBindable` to an element that does match the `appHighlight` directive, the highlight starts to work again.

So, it seems as if `ngNonBindable` does indeed prevent directives from being applied to descendant elements, but not the element itself. Also, property bindings `[name]="name"` and interpolated attributes `name="{{ name }}"` still work on an element that is marked `ngNonBindable`, but do no longer work within the descendant tree of an element marked `ngNonBindable`. In conclusion, `ngNonBindable` only affects descendants and not the element itself.

Thanks a lot for your experiment! Hopefully the above makes it more clear and allows you to incorporate (some of) these details in the documentation.

Comment (kapunahelewong) [reply aio/content/guide/attribute-directives.md]: Ohhhh, this is so helpful! Thank you, @JoostK! I didn't realize that but now that I get it, it's a really important point. Wow, I'm going to add this to my example and figure out how to incorporate your explanation. You made my day! :)

**Comment by kapunahelewong** [comment]
Holding this PR until Template Syntax PRs merge. 

Binding Syntax: https://github.com/angular/angular/pull/25561
Property Binding: https://github.com/angular/angular/pull/25770
Attribute Binding: https://github.com/angular/angular/pull/26004
Two-way binding: https://github.com/angular/angular/pull/26278
Built-in directives: https://github.com/angular/angular/pull/27273
Template Ref Vars: https://github.com/angular/angular/pull/27371
Inputs Outputs: https://github.com/angular/angular/pull/27685
Template Expression Operators: https://github.com/angular/angular/pull/28087

After these are merged, Template Syntax will be separated into different docs. At that point, we can find the best place for this.

**Comment by ajitsinghkaler** [comment]
@kapunahelewong all these pr has been merged so we can start merging this pr if you need any help please tell me. Or if I can work on this pr.

**Comment by kapunahelewong** [comment]
Thank you, @ajitsinghkaler! If you'd like, you could take this PR over. It will still need to be reviewed by engineering and docs to make sure it's in accordance with what should be in the documentation about it as well as that it is technically correct. If you take it, please tag me and reference the PR and I will close this one out and refer to your PR. 🌈

**Comment by ajitsinghkaler** [comment]
Would love to please assign it to me

**Comment by ajitsinghkaler** [comment]
@kapunahelewong you can close this in favor of #36560

**Comment by kapunahelewong** [comment]
Very kind of you! Thanks a bunch, @ajitsinghkaler!

**Comment by ajitsinghkaler** [comment]
No problem

**Pull request closed without merging**